runtime.go

```diff
@@ -2216,10 +2277,13 @@ func AssertFunction(v Value) (Callable, bool) {
 			return func(this Value, args ...Value) (ret Value, err error) {
 				defer func() {
 					if x := recover(); x != nil {
+						// TODO: Are these two error handling supposed to live side by side ?
 						if ex, ok := x.(*InterruptedError); ok {
 							if ex.Error() != "" && ex.Error() != "<nil>" {
 								err = ex
 							}
+						} else if ex, ok := x.(*uncatchableException); ok {
```

Comment:
Not too sure about the addition of this `else if`, in the `master` branch they only have the `uncatchableException`.

Reply:
The original [vm.run](https://github.com/dop251/goja/blob/master/vm.go#L387) function no longer panics with the `InterruptedError`. I would lean towards aligning this function to the original one by getting rid of the `InterruptedError` handling in here and avoiding to panic with it in the `run` function.

Note that whatever changes we make to this function will likely need to be applied to `ToFunctionWithContext` as well.